addons/pos_sale/models/sale_order.py

```diff
@@ -82,3 +82,8 @@ def _convert_qty(self, sale_line, qty, direction):
             return sale_line_uom._compute_quantity(qty, product_uom, False)
         elif direction == 'p2s':
             return product_uom._compute_quantity(qty, sale_line_uom, False)
+
+    def unlink(self):
+        # do not delete downpayment lines created from pos
+        pos_downpayment_lines = self.filtered(lambda line: line.is_downpayment and line.pos_order_line_ids)
```

Comment:
add `sudo()` if user have not access to pos_order_line_ids
`pos_downpayment_lines = self.sudo().filtered(lambda line: line.is_downpayment and line.pos_order_line_ids)`